vowpalwabbit/cbify.cc

```diff
@@ -97,6 +119,43 @@ float loss_csldf(cbify& data, std::vector<v_array<COST_SENSITIVE::wclass>>& cs_c
   return data.loss0 + (data.loss1 - data.loss0) * cost;
 }
 
+template <class T>
+inline void delete_it(T* p)
+{
+  if (p != nullptr)
+    delete p;
+}
+
+void finish_cbify_reg(cbify_reg& data, std::ostream* trace_stream)
+{
+  if (trace_stream != nullptr)
+    (*trace_stream) << "Max Cost=" << data.max_cost << std::endl;
+
+  data.cb_cont_label.costs.delete_v();  // todo: instead of above
+}
+
+void finish(cbify& data)
+{
+  CB::cb_label.delete_label(&data.cb_label);
+  data.a_s.delete_v();
+  finish_cbify_reg(data.regression_data, &data.all->trace_message);
+
+  if (data.use_adf)
+  {
+    for (size_t a = 0; a < data.adf_data.num_actions; ++a)
+    {
+      data.adf_data.ecs[a]->pred.a_s.delete_v();
+      VW::dealloc_example(CB::cb_label.delete_label, *data.adf_data.ecs[a]);
+      free_it(data.adf_data.ecs[a]);
+    }
+    data.adf_data.ecs.~vector<example*>();
```

Comment:
Why are all of these explicit destructors requried?

Reply:
This is not being used.  Removed.